jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/gcprules/vminstance/VMInstanceSerialPortDisableRule.java

```diff
@@ -0,0 +1,132 @@
+        String severity = ruleParam.get(PacmanRuleConstants.SEVERITY);
+        String category = ruleParam.get(PacmanRuleConstants.CATEGORY);
+
+        String vmEsURL = CommonUtils.getEnvVariableValue(PacmanSdkConstants.ES_URI_ENV_VAR_NAME);
+
+        if (Boolean.FALSE.equals(PacmanUtils.doesAllHaveValue(severity, category, vmEsURL))) {
+            logger.info("{},{},{}",severity,category,vmEsURL);
+            logger.info(PacmanRuleConstants.MISSING_CONFIGURATION);
+            throw new InvalidInputException(PacmanRuleConstants.MISSING_CONFIGURATION);
+        }
+
+        if (!StringUtils.isNullOrEmpty(vmEsURL)) {
+            vmEsURL = vmEsURL + "/gcp_vminstance/_search";
+        }
+        logger.debug("========vmEsURL URL after concatenation param {}  =========", vmEsURL);
+
+        boolean isBlockProjectWideSSHEnable = false;
+
+        MDC.put("executionId", ruleParam.get("executionId"));
+        MDC.put("ruleId", ruleParam.get(PacmanSdkConstants.RULE_ID));
+
+        if (!StringUtils.isNullOrEmpty(resourceId)) {
+
+            Map<String, Object> mustFilter = new HashMap<>();
+            mustFilter.put(PacmanUtils.convertAttributetoKeyword(PacmanRuleConstants.RESOURCE_ID), resourceId);
+            mustFilter.put(PacmanRuleConstants.LATEST, true);
+
+            try {
+                isBlockProjectWideSSHEnable = verifyIsBlockProjectWideSSHEnable(vmEsURL, mustFilter);
+                if (!isBlockProjectWideSSHEnable) {
+                    List<LinkedHashMap<String, Object>> issueList = new ArrayList<>();
+                    LinkedHashMap<String, Object> issue = new LinkedHashMap<>();
+
+                    annotation = Annotation.buildAnnotation(ruleParam, Annotation.Type.ISSUE);
+                    annotation.put(PacmanSdkConstants.DESCRIPTION, "serial-port-enable key is enabled");
+                    annotation.put(PacmanRuleConstants.SEVERITY, severity);
+                    annotation.put(PacmanRuleConstants.CATEGORY, category);
+
+                    issue.put(PacmanRuleConstants.VIOLATION_REASON, "serial-port-enable key is enabled");
+                    issueList.add(issue);
+                    annotation.put("issueDetails", issueList.toString());
+                    logger.debug("========VMInstanceSerialPortDisableRule ended with an annotation {} : =========", annotation);
+                    return new RuleResult(PacmanSdkConstants.STATUS_FAILURE, PacmanRuleConstants.FAILURE_MESSAGE, annotation);
+                }
+
+            } catch (Exception exception) {
+                throw new RuleExecutionFailedExeption(exception.getMessage());
+            }
+        }
+
+        logger.debug("========VMInstanceSerialPortDisableRule ended=========");
+        return new RuleResult(PacmanSdkConstants.STATUS_SUCCESS, PacmanRuleConstants.SUCCESS_MESSAGE);
+
+    }
+
+    private boolean verifyIsBlockProjectWideSSHEnable(String vmEsURL, Map<String, Object> mustFilter) throws Exception {
```

Comment:
Can we rename the method relevant to the policy name .

Reply:
changed method name to verifyIsSerialPortDisable( ).